src/psyclone/tests/psyir/symbols/symboltable_test.py

```diff
@@ -1643,3 +1646,232 @@ def test_rename_symbol():
         schedule_symbol_table.rename_symbol(symbol, "aRRay")
     assert ("The name argument of rename_symbol() must not already exist in "
             "this symbol_table instance, but 'aRRay' does." in str(err.value))
+
+
+def test_resolve_imports(fortran_reader, tmpdir):
+    ''' Tests the SymbolTable resolve_imports method. '''
+    with open(os.path.join(str(tmpdir), "a_mod.f90"), "w") as module:
+        module.write('''
+        module a_mod
+            integer :: a_1, a_2
+            integer :: b_1  ! Name clash but it is not imported
+        end module a_mod
+        ''')
+    with open(os.path.join(str(tmpdir), "b_mod.f90"), "w") as module:
+        module.write('''
+        module b_mod
+            integer, parameter :: b_1 = 10
+            integer, save, pointer :: b_2
+            integer :: not_used1
+            integer :: not_used2
+        end module b_mod
+        ''')
+    psyir = fortran_reader.psyir_from_source('''
+        module test_mod
+            use a_mod, only: a_2
+            private :: a_2
+            contains
+            subroutine test()
+                use a_mod, only: a_1
+                use b_mod
+                use c_mod  ! This module is not found in INCLUDE_PATH
+
+                a_1 = b_1 + b_2
+            end subroutine test
+        end module test_mod
+    ''')
+    subroutine = psyir.walk(Routine)[0]
+    # Add Generic unresolved reference to "not_used"
+    subroutine.symbol_table.add(
+            Symbol("not_used1", interface=ImportInterface(
+                subroutine.symbol_table.lookup("b_mod"))))
+
+    # After parsing the a_1, a_2, b_1, b_2, not_used1 and not_used2 will have#
+    # incomplete information because the modules are not resolved
+    a_1 = subroutine.symbol_table.lookup('a_1')
+    a_2 = subroutine.symbol_table.lookup('a_2')
+    b_1 = subroutine.symbol_table.lookup('b_1')
+    b_2 = subroutine.symbol_table.lookup('b_2')
+    not_used1 = subroutine.symbol_table.lookup('not_used1')
+    assert "not_used2" not in subroutine.symbol_table
+    assert not isinstance(not_used1, DataSymbol)
+    assert isinstance(a_1.interface, ImportInterface)
+    assert not isinstance(a_1, DataSymbol)
+    assert isinstance(a_2.interface, ImportInterface)
+    assert not isinstance(a_2, DataSymbol)
+    assert a_2.visibility == Symbol.Visibility.PRIVATE
+    assert isinstance(b_1.interface, UnresolvedInterface)
+    assert not isinstance(b_1, DataSymbol)
+    assert isinstance(b_2.interface, UnresolvedInterface)
+    assert not isinstance(b_2, DataSymbol)
+
+    # Set up include_path to import the proper modules
+    Config.get()._include_paths = [str(tmpdir)]
```

Comment:
Probably should use monkeypatch here to avoid polluting other tests.

Reply:
Oh yes, done!